compiler/compiler.go

```diff
@@ -168,11 +168,8 @@ func WriteProgramCode(pkgs []*Archive, w *SourceMapFilter) error {
 		}
 	}
 
-	if _, err := w.Write([]byte("$synthesizeMethods();\nvar $mainPkg = $packages[\"" + string(mainPkg.ImportPath) + "\"];\n$packages[\"runtime\"].$init();\n$go($mainPkg.$init, []);\n$flushConsole();\n\n}).call(this);\n")); err != nil {
-		return err
-	}
-
-	return nil
+	_, err := w.Write([]byte("$synthesizeMethods();\nvar $mainPkg = $packages[\"" + string(mainPkg.ImportPath) + "\"];\n$packages[\"runtime\"].$init();\n$go($mainPkg.$init, []);\n$flushConsole();\n\n}).call(this);\n"))
```

Comment:
Up until about a year ago, I used to feel the same way. But I saw that the Go standard library did it, and I decided to try it after all. Then this simplification grew on me.

Now, when I read Go code, seeing `return err` at the end of a function is absolutely equivalent and interchangeable with `if err != nil { return err }; return nil`, except it avoids being unnecessarily longer.

I've also started to not mind diffs. Unless the code happens to be very very WIP, diffs are actually pretty rare. So I'd rather look at the shorter equivalent version for a year before a change, and when a chance is neccessary, the diff is not at all that much worse (i.e., I find it very readable too).

In some places, I still prefer to do `err := DoSomething(); return err` instead of just `return DoSomething()`, when I want to emphasize that doing the thing is more important than returning its error value. But even that is almost interchangeable for me now, and that's why I wrote `return s.BuildFiles(args, pkgObj, currentDirectory)` below, instead of `err = s.BuildFiles(args, pkgObj, currentDirectory); return err`.

But okay, I will back out this change for now since you prefer the `if err != nil { return err }; return nil` version and we can always change it later if you change your mind.

(/cc @dominikh FYI, this is related to S1013.)